tests/intmap-properties.hs

```diff
@@ -28,6 +31,16 @@ import Test.QuickCheck.Function (Fun(..), apply)
 
 default (Int)
 
+newtype Consty a b = Consty { getConsty :: a}
+  deriving (Eq)
+
+instance Functor (Consty a) where
+  fmap _ (Consty a) = Consty a
+
+instance Monoid m => Applicative (Consty m) where
+  pure = const (Consty mempty)
+  (Consty l) <*> (Consty r) = Consty (l `mappend` r)
+
```

Comment:
Why do we need this? Why not use `Control.Applicative.Const`? For that matter, `traverse_` is defined in terms of `foldr`. So the `prop_traverseEqTraverse_` test is really about making sure the `Foldable` and `Traversable` instances are compatible. A very direct way to do that is to use `foldMapDefault`, something like this:

```haskell
prop_FoldableTraversableCompat :: Fun A [B] -> IntMap A -> Property
prop_FoldableTraversableCompat fun m = foldMap f m === foldMapDefault f m
  where f = apply fun
```

Reply:
> Why not use Control.Applicative.Const?

I was being consistent with map-properties.hs, which defines its own `Consty`, apparently by you in https://github.com/haskell/containers/commit/7d03d76b7647dc731da2e75dcdd45f6c2e667682 :) Not sure what the original rationale was, maybe because Const needed a separate package at some point in time? /shrug

 >  A very direct way to do that.... 

Great observation, I agree that's much better than relying on a specific applicative instance. I've used `String` and `Int` in place of `A` and `B`, let me know if you had something else in mind.

Reply:
`A` and `B` come from `Test.QuickCheck.Poly` or something. They're really `Integer` under the hood, but they make it clear we don't care.

Reply:
I dunno why I used `Consty` for `map-properties` either... Hrmmmmm.....